### PR TITLE
LPS-160039 Patch Freemarker to not use Introspector when filling propertySetters map

### PR DESCRIPTION
**Pull request by guilhermedcamacho**

### CODE REVIEW

**Comment by guilhermedcamacho** [comment]
ci:test:commerce

**Comment by guilhermedcamacho** [comment]
ci:test:acceptance

**Comment by guilhermedcamacho** [comment]
@catarineaguiar please help me check the results, thanks!

**Comment by catarineaguiar** [comment]
I'm waiting for the testray results.

**Comment by catarineaguiar** [comment]
@guilhermedcamacho no regressions found.

**Comment by guilhermedcamacho** [comment]
@shuyangzhou , this PR is ready for your review. Thanks!

**Comment by guilhermedcamacho** [comment]
I forced-pushed to fix the conflict of file `lib/versions-complete.xml`

File: modules/third-party/org-freemarker/patches/LPS-160230.patch [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+-        PropertyDescriptor[] pda = bi.getPropertyDescriptors();
+-        for (int i = 0; i < pda.length; i++) {
+-            PropertyDescriptor pd = pda[i];
+-            Method m = pd.getWriteMethod();
+-            if (m != null) {
+-                propertySetters.put(pd.getName(), m);
++        for (Method method : tagClass.getMethods()) {
++            if (method.getReturnType().equals(void.class) &&
++                (method.getParameterTypes().length == 1) &&
++                method.getName().matches("^set[A-Z].*")) {
```
Comment (shuyangzhou) [inline modules/third-party/org-freemarker/patches/LPS-160230.patch:34]: @guilhermedcamacho please get rid of this regex usage, just do plain String startWith and length testing. Also please adjust the commit order. The patching commit should go right after the initial checkin.

**Comment by shuyangzhou** [comment]
See above comment, please resend.

**Pull request closed without merging**